yt/frontends/tipsy/data_structures.py

```diff
@@ -122,7 +126,9 @@ def _parse_parameter_file(self):
         hvals = {
             a: c
             for (a, b), c in zip(
-                self._header_spec, struct.unpack(hh, f.read(struct.calcsize(hh)))
+                self._header_spec,
```

Comment:
This one I'm less certain of, in that it may be the case that we don't want to force the specification of all fields and just let some go.  But, that's not necessarily what we should build in to doing this check, so :+1: .